framework/smoke_test_framework.cpp

```diff
@@ -0,0 +1,592 @@
+      totalTasks((_totalTasks.isSome()?_totalTasks.get():1)),
+      targetHostname(_targetHostname),
+      tasksLaunched(0u),
+      scheduled(false) {}
+
+  const string command;
+  const Resources taskResources;
+  const size_t totalTasks;
+  size_t tasksLaunched;
+  bool scheduled;
+
+  const Option<string> targetHostname;
+
+  void print() {
+    LOG(INFO) << "| Command: " << this->command
+        << "; Resources: " <<  this->taskResources
+        << "; Tasks to spawn: " << this->totalTasks
+        << "; Target hostname: "
+        << (this->targetHostname.isSome()?this->targetHostname.get():"<all>")
+        << "|";
+
+  }
+};
+
+
+class SerenityNoExecutorScheduler : public Scheduler
+{
+public:
+  SerenityNoExecutorScheduler(
+      const FrameworkInfo& _frameworkInfo,
+      const list<JobSpec>& _jobs)
+    : frameworkInfo(_frameworkInfo),
+      jobs(_jobs),
+      tasksFinished(0u),
+      tasksTerminated(0u),
+      tasksTodo(0u),
+      jobScheduled(0u) {
+    foreach (const JobSpec& job, jobs) {
+      tasksTodo += job.totalTasks;
+    }
+  }
+
+  virtual void registered(
+      SchedulerDriver* driver,
+      const FrameworkID& frameworkId,
+      const MasterInfo& masterInfo)
+  {
+    LOG(INFO) << "Registered with master " << masterInfo
+              << " and got framework ID " << frameworkId;
+
+    frameworkInfo.mutable_id()->CopyFrom(frameworkId);
+  }
+
+  virtual void reregistered(
+      SchedulerDriver* driver,
+      const MasterInfo& masterInfo)
+  {
+    LOG(INFO) << "Reregistered with master " << masterInfo;
+  }
+
+  virtual void disconnected(
+      SchedulerDriver* driver)
+  {
+    LOG(INFO) << "Disconnected!";
+  }
+
+  virtual void resourceOffers(
+      SchedulerDriver* driver,
+      const vector<Offer>& offers)
+  {
+    Filters filters;
+    filters.set_refuse_seconds(Duration::max().secs());
+    foreach (const Offer& offer, offers) {
+      if (jobScheduled >= jobs.size()) {
+        driver->declineOffer(offer.id(), filters);
+        continue;
+      }
+      LOG(INFO) << "Received offer " << offer.id() << " from slave "
+                << offer.slave_id() << " (" << offer.hostname() << ") "
+                << "with " << offer.resources();
+
+      Resources remaining = offer.resources();
+      vector<TaskInfo> tasks;
+      int i = 0;
+      for(auto job=jobs.begin(); job != jobs.end(); ++job, ++i) {
+        if (job->scheduled) continue;
+        if (job->targetHostname.isSome() &&
+            job->targetHostname.get().compare(offer.hostname()) != 0){
+          LOG(INFO) << "Hosts not matched! Omitting.";
+          continue;
+        }
+
+        while (true) {
+          if (!remaining.contains(job->taskResources)) {
+            LOG(INFO) << "Not enough resources for "
+                      << stringify(i) + "_" + stringify(job->tasksLaunched)
+                      << " job. Needed: " << job->taskResources
+                      << " Offered: " << remaining;
+            break;
```

Comment:
I don't think so. It is done in same way it is done currently in _no executor framework_. When there are not enough resources then i just 'accept' this offer with no _task_ to launch in `tasks` vector. See `driver->acceptOffers(...` below.